tests/ns1_team.yaml

```diff
@@ -0,0 +1,187 @@
+---
+- hosts: localhost
+  tasks:
+    ##########
+    # Verify ansible test team is absent before starting.
+    ##########
+    - name: setup
+      local_action:
+        module: ns1_team
+        apiKey: "{{ ns1_token }}"
+        name: "{{ test_team }}"
+        state: absent
+      register: team_setup
+
+    - name: Verify setup
+      assert:
+        that:
+          - team_setup is success
+
+    ##########
+    # Test team creation
+    ##########
+    - name: Test team creation
+      local_action:
+        module: ns1_team
+        apiKey: "{{ ns1_token }}"
+        name: "{{ test_team }}"
+        state: present
+      register: team_create
+
+    - name: Verify team creation
+      assert:
+        that:
+          - team_create is changed
+          - team_create.diff.after.name == "{{ test_team }}"
+
+    ##########
+    # Test team ip whitelist update
+    ##########
+    - name: Test team ip whitelist
+      local_action:
+        module: ns1_team
+        apiKey: "{{ ns1_token }}"
+        name: "{{ test_team }}"
+        state: present
+        ip_whitelist:
+          - name: Home Whitelist
+            values:
+            - 104.20.48.182
+          - name: Multi Whitelist
+            values:
+            - 104.20.49.0/24
+            - 104.20.50.1
+      register: team_ip_whitelist
+
+    - name: Verify team ip whitelist
+      assert:
+        that:
+          - team_ip_whitelist is changed
+          - team_ip_whitelist.diff.after.ip_whitelist | length != 0
+
+    ##########
+    # Test team permissions update: monitoring
+    ##########
+    - name: Add all monitoring permissions
+      local_action:
+        module: ns1_team
+        apiKey: "{{ ns1_token }}"
+        name: "{{ test_team }}"
+        state: present
+        permissions:
+          monitoring:
+            manage_jobs: true
+            view_jobs: true
+            manage_lists: true
+      register: team_monitoring
+
+    - name: Verify permissions are set
+      assert:
+        that:
+          - team_monitoring is changed
```

Comment:
Ditto, would be good to assert on the values that were updated to ensure the correct changes were made.